src/ParquetMsg.chpl

```diff
@@ -796,19 +774,12 @@ module ParquetMsg {
   proc lspqMsg(cmd: string, payload: string, st: borrowed SymTab): MsgTuple throws {
     // reqMsg: "lshdf [<json_filename>]"
     var repMsg: string;
-    var (jsonfile) = payload.splitMsgToTuple(1);
+    var msgArgs = parseMessageArgs(payload, 1);
 
     // Retrieve filename from payload
-    var filename: string;
-    try {
-      filename = jsonToPdArray(jsonfile, 1)[0];
-      if filename.isEmpty() {
-        throw new IllegalArgumentError("filename was empty");  // will be caught by catch block
-      }
-    } catch {
-      var errorMsg = "Could not decode json filenames via tempfile (%i files: %s)".format(
-                                                                                          1, jsonfile);
-      return new MsgTuple(errorMsg, MsgType.ERROR);                                    
+    var filename: string = msgArgs.getValueOf("filename");
+    if filename.isEmpty() {
+      throw new IllegalArgumentError("filename was empty");  // will be caught by catch block
```

Comment:
should we return error message tuple instead?

Reply:
Updated.